core/src/test/kotlin/generator/test/GeneratorTest.kt

```diff
@@ -0,0 +1,38 @@
+package generator.test
+
+import generator.TestRunner
+import org.jetbrains.dokka.DokkaTestGenerator
+import org.jetbrains.dokka.utilities.DokkaConsoleLogger
+import org.junit.Test
+
+class GeneratorTest {
+
+    val logger = DokkaConsoleLogger
+
+    @Test
+    fun test1() {
+        val checks: Map<DokkaTestGenerator.Companion.Stage, (DokkaTestGenerator.State) -> Boolean> = mapOf(
+            DokkaTestGenerator.Companion.Stage.DOCUMENTABLES_CREATION to { state: DokkaTestGenerator.State ->
+                state.context?.platforms?.forEach { (pd, _) ->
+                    logger.info(pd.name)
+                }
+                true
+            },
+            DokkaTestGenerator.Companion.Stage.RENDERING to { state: DokkaTestGenerator.State ->
+                val test = state.pages?.parentMap?.size == 7
+                assert(test)
+                test
```

Comment:
I think the assert should be lifted from here as we're returning the test result up

Reply:
I've changed return type to Unit? because in my opinion it's better to keep asserts inside tests.